homeassistant/components/mobile_app/__init__.py

```diff
@@ -115,3 +118,15 @@ class MobileAppFlowHandler(config_entries.ConfigFlow):
         """Handle a flow initialized during registration."""
         return self.async_create_entry(title=user_input[ATTR_DEVICE_NAME],
                                        data=user_input)
+
+
+def push_registrations(hass):
+    """Return a dictionary of push enabled registrations."""
+    targets = {}
+    for webhook_id, entry in hass.data[DOMAIN][DATA_CONFIG_ENTRIES].items():
+        data = entry.data
+        app_data = data[ATTR_APP_DATA]
+        if ATTR_PUSH_TOKEN in app_data and ATTR_PUSH_URL in app_data:
+            device_name = data[ATTR_DEVICE_NAME]
+            targets[device_name] = webhook_id
```

Comment:
You are using device name as a key but that implies that they need to be unique. That's not the case. `webhook_id` is unique.

Reply:
True, but the primary purpose of this function is to provide a dictionary that can be used in the `notify.targets` function. The key needs to be a human readable string. I don't think a webhook ID is that. Even if I fix this dictionary specifically, its still an issue for targets where the name is duplicative of an existing target, so wat do?

Reply:
Well first, don't put anything in `mobile_app/__init__.py` that is meant to provide data just for notify.targets 😉 

Would it help if we allow targets to be specified as device tracker entity IDs?

Reply:
hm, although that makes location reporting required.

Reply:
To be fair, this was also an issue on the `ios.notify` platform too.

Reply:
Well, we can always just add a device tracker entity and just set state to `unknown` . That way there is still an entity ID .